tests/core/runners/test_ase.py

```diff
@@ -47,6 +52,196 @@ def teardown_function():
+            [0.8409564131524673, 0.018549610257914483, -0.07338809662321308],
+            [-0.6399757891931867, 0.01763740851518944, 0.0581573443268891],
+            [-1.0005576455546672, 1.0430257532387608, 0.22197240310602892],
+            [1.402180736662139, 0.944112416574632, -0.12179540364365492],
+            [-1.1216961389434357, -0.3883639833876232, -0.8769102842015071],
+            [-0.9645026578514683, -0.6204201840686793, 0.9240543090678239],
+        ],
+    )
+
+    product = Atoms(
+        symbols="CCHHCHH",
+        positions=[
+            [1.348003553501624, 0.4819311116778978, 0.2752537177143993],
+            [0.2386618286631742, -0.3433222966734429, 0.37705518940917926],
+            [-0.9741307940518336, 0.07686022294949588, 0.08710778043683955],
+            [-1.8314843503320921, -0.5547344604780035, 0.1639037492534953],
+            [0.3801391040059668, -1.3793340533058087, 0.71035902765307],
+            [1.9296265384257907, 0.622088341468767, 1.0901733942191298],
+            [-1.090815880212625, 1.0965111343610956, -0.23791518420660265],
+        ],
+    )
+    return reactant, product
+
+
+@pytest.mark.parametrize(
+    (
+        "nimages",
+        "convergence_tolerance",
+        "max_iterations",
+        "max_micro_iterations",
+        "morse_scaling",
+        "geometry_friction",
+        "distance_cutoff",
+        "save_raw_path",
+        "expected_length",
+    ),
+    [
+        (20, 2e-3, 15, 20, 1.7, 1e-2, 3.0, None, 20),  # Default parameters
+        (10, 2e-3, 15, 20, 1.7, 1e-2, 3.0, None, 10),  # Different number of images
+        (
+            20,
+            1e-4,
+            10,
+            10,
+            1.5,
+            0.01,
+            2.5,
+            "raw_path.xyz",
+            20,
+        ),  # Different interpolation parameters and save_raw
+    ],
+)
+def test_geodesic_interpolate_wrapper(
+    setup_test_environment,
+    nimages,
+    convergence_tolerance,
+    max_iterations,
+    max_micro_iterations,
+    morse_scaling,
+    geometry_friction,
+    distance_cutoff,
+    save_raw_path,
+    expected_length,
+):
+    reactant, product = setup_test_environment
+    # Execute the geodesic_interpolate_wrapper function
+    smoother_path = _geodesic_interpolate_wrapper(
+        reactant,
+        product,
+        nimages=nimages,
+        convergence_tolerance=convergence_tolerance,
+        max_iterations=max_iterations,
+        max_micro_iterations=max_micro_iterations,
+        morse_scaling=morse_scaling,
+        geometry_friction=geometry_friction,
+        distance_cutoff=distance_cutoff,
+    )
+    assert smoother_path[1].positions[0][0] == pytest.approx(1.36055556030, abs=1e-1)
+
+
+def test_geodesic_interpolate_wrapper_large_system(setup_test_environment):
+    rng = np.random.default_rng()  # Create a random number generator instance
+    large_atoms = Atoms("H" * 40, positions=rng.random((40, 3)))
```

Comment:
We do not want to introduce randomness into our test suite. Please define an `Atoms` object that is deterministic.

Reply:
This is because the results from Geodesic are not deterministic.
This is a serious limitation for testing.
@Andrew-S-Rosen @samblau 

There are two places in geodesic code where I found the use of random number generators. I can try and make it have a seed so that the results are reproducible. I have tried very quickly but not succeeded so far.

Reply:
Even if geodesic-interpolate is stochastic, why would we need a random `Atoms` object to provide?

Reply:
I have replaced it with C60. Thanks for the suggestion.